rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/dataspace/RestDataspaceImpl.java

```diff
@@ -400,20 +420,26 @@ private Response serverErrorRes(String format, Object... args) {
                 (args == null || args.length == 0) ? format : String.format(format, args)).build();
     }
 
-    private FileObject resolveFile(Session session, String dataspace, String pathname)
-            throws FileSystemException, NotConnectedException, PermissionException {
-        return "user".equals(dataspace) ? fileSystem(session).resolveFileInUserspace(pathname) : fileSystem(
-                session).resolveFileInGlobalspace(pathname);
+    public FileObject resolveFile(Session session, String dataspace, String pathname)
+            throws FileSystemException, NotConnectedRestException, PermissionRestException {
+        try {
+            return USER.equalsIgnoreCase(dataspace) | SchedulerConstants.USERSPACE_NAME.toString().equalsIgnoreCase(dataspace) ? fileSystem(session).resolveFileInUserspace(pathname) : fileSystem(
+                    session).resolveFileInGlobalspace(pathname);
+        } catch (NotConnectedException e) {
+            throw new NotConnectedRestException(e);
+        } catch (PermissionException e) {
+            throw new PermissionRestException(e);
+        }
     }
 
     private void checkPathParams(String dataspace, String pathname) {
         checkArgument(!Strings.isNullOrEmpty(dataspace), "Dataspace name cannot be null or empty.");
-        checkArgument("user".equals(dataspace) || "global".equals(dataspace),
-                "Invalid dataspace name: '%s', only 'user' or 'global' is allowed.", dataspace);
+        checkArgument(USER.equalsIgnoreCase(dataspace) || GLOBAL.equalsIgnoreCase(dataspace) || SchedulerConstants.USERSPACE_NAME.toString().equalsIgnoreCase(dataspace) || SchedulerConstants.GLOBALSPACE_NAME.toString().equalsIgnoreCase(dataspace),
+                "Invalid dataspace name: '%s'.", dataspace);
         checkArgument(!Strings.isNullOrEmpty(pathname), "Pathname cannot be null or empty.");
     }
 
-    private void writeFile(InputStream inputStream, FileObject outputFile, String encoding)
+    public void writeFile(InputStream inputStream, FileObject outputFile, String encoding)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: org.apache.commons.vfs2.FileSystemException, java.io.IOException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1160)